### PR TITLE
added to ExpandableLayout a few methods.  

### PR DESCRIPTION
**Pull request by sd6352051**
`toggle(boolean animated)`
`expand(boolean animated)`
`collapse(boolean animated)`

Set whether this view animated to extend

`true` to animated to extend, `false` to transition immediately

In some cases .I don't need this animation.
such as when view has been scrolled

### CODE REVIEW

File: lib/src/main/java/net/cachapa/expandablelayout/ExpandableLinearLayout.java [hunk 1]
```diff
@@ -178,15 +186,19 @@ public void expand() {
         }
 
         for (View expandableView : expandableViews) {
-            animateHeight(expandableView, expandableView.getMeasuredHeight());
+            animateHeight(expandableView, expandableView.getMeasuredHeight(),animated);
```
Comment (cachapa) [inline lib/src/main/java/net/cachapa/expandablelayout/ExpandableLinearLayout.java]: Code style: please add space after comma: ", animated"

File: lib/src/main/java/net/cachapa/expandablelayout/ExpandableLinearLayout.java [hunk 2]
```diff
@@ -147,15 +147,23 @@ public boolean isExpanded() {
     }
 
     public void toggle() {
+        toggle(true);
+    }
+
+    public void toggle(boolean animated) {
```
Comment (cachapa) [inline lib/src/main/java/net/cachapa/expandablelayout/ExpandableLinearLayout.java]: Code style: please use "animate" instead of "animated".
This also applies to all other methods

File: lib/src/main/java/net/cachapa/expandablelayout/ExpandableLinearLayout.java [hunk 3]
```diff
@@ -211,7 +223,7 @@ public void setOnExpansionUpdateListener(OnExpansionUpdateListener listener) {
         this.listener = listener;
     }
 
-    private void animateHeight(final View view, final int targetHeight) {
+    private void animateHeight(final View view, final int targetHeight,boolean animated) {
```
Comment (cachapa) [inline lib/src/main/java/net/cachapa/expandablelayout/ExpandableLinearLayout.java]: Code style: please add space after comma: ", animated"

Comment (cachapa) [inline lib/src/main/java/net/cachapa/expandablelayout/ExpandableLinearLayout.java]: I'm not a fan of return statements in the middle of methods, as it makes it hard to follow the code path.
The non-animated option should be separated into its own function.

Comment (sd6352051) [inline lib/src/main/java/net/cachapa/expandablelayout/ExpandableLinearLayout.java]: Yes, you're right.
this is wrong.
I will modify it

File: lib/src/main/java/net/cachapa/expandablelayout/ExpandableLinearLayout.java [hunk 4]
```diff
@@ -222,6 +234,24 @@ private void animateHeight(final View view, final int targetHeight) {
         lp.weight = 0;
         int height = view.getHeight();
 
+        if (!animate) {
```
Comment (cachapa) [inline lib/src/main/java/net/cachapa/expandablelayout/ExpandableLinearLayout.java]: I'm not a fan of return statements in the middle of methods, as it makes it hard to follow the code path.
The non-animated option should be separated into its own function.

Since I was meaning to refactor this code anyhow, I'm going to accept this pull request and refactor on top of it.

**Comment by cachapa** [comment]
Thanks for the pull request. It's a useful addition to the library.

**Comment by sd6352051** [comment]
Great lib.You are the man ;)
